app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/vm/operations.rb

```diff
@@ -15,4 +15,30 @@ def raw_create_snapshot(name, desc = nil, _memory = nil)
     create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "create")
     raise MiqException::MiqVmSnapshotError, err.to_s
   end
+
+  def raw_remove_snapshot(snapshot_id)
+    with_provider_connection(:service => 'PCloudSnapshotsApi') do |api|
+      snapshot = Snapshot.find_by(:id => snapshot_id)
+      api.pcloud_cloudinstances_snapshots_delete(cloud_instance_id, snapshot.uid_ems)
+    end
+  rescue => err
+    create_notification(:vm_snapshot_failure, :error => err.to_s, :snapshot_op => "delete")
+    raise MiqException::MiqVmSnapshotError, err.to_s
+  end
+
+  def raw_remove_all_snapshots
+    with_provider_connection(:service => 'PCloudSnapshotsApi') do |api|
+      vm = Vm.find_by(:uid_ems => ems_ref)
+      snapshots = Snapshot.where(:vm_or_template_id => vm.id)
+      snapshots.each do |snapshot|
```

Comment:
You should already have a `snapshots` association defined on the vm so this should work without the previous two queries

Reply:
Wonderful. This is great. The less code, the better! Tested working without the 2 lines, and force-pushed.